src/Server/HTTPHandler.cpp

```diff
@@ -760,76 +759,84 @@ void HTTPHandler::processQuery(
 
     /// Send HTTP headers with code 200 if no exception happened and the data is still not sent to
     /// the client.
-    used_output.out->finalize();
+    used_output.finalize();
 }
 
 void HTTPHandler::trySendExceptionToClient(
     const std::string & s, int exception_code, HTTPServerRequest & request, HTTPServerResponse & response, Output & used_output)
+try
 {
-    try
+    response.set("X-ClickHouse-Exception-Code", toString<int>(exception_code));
+
+    /// FIXME: make sure that no one else is reading from the same stream at the moment.
```

Comment:
What will happen if there are multiple readers?

Reply:
Let's ask @abyss7 (the author of this `FIXME`) ?